src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -1310,6 +1357,102 @@ static status_t del_routes_internal(private_kernel_super_ipsec_t* this,
 }
 
 
+METHOD(kernel_ipsec_t, write_terminator_tunnel_ike_sa_uptime, status_t,
+	private_kernel_super_ipsec_t* this,
+	ike_sa_t* ike_sa)
+{
+	(void)this;
+	const ike_super_peer_info_t* super_peer_info = ike_sa->get_super_peer_info(ike_sa);
+	if(! super_peer_info) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get super peer info");
+		return FAILED;
+	}
+
+	if(IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
+		return NOT_SUPPORTED;
+	}
+	const char *vrf_name = super_peer_info->get_vrf_name(super_peer_info);
+	const char *device_id = super_peer_info->get_device_id(super_peer_info);
```

Comment:
In regards to the fact that this should be written for `IKE_SUPER_PEER_INFO_TYPE_TUNNEL`, in case of configured-tunnel both `vrf_name` and `device_id` will not be set in the `super_peer_info`.
Currently the only identifier of a configured-tunnel is the tunnel_id (a mix of the `vrf_name` and `device_id` of the tunnel), stored in peer_cfg (see `peer_cfg->get_super_tunnel_id()`).
If you need the device_id/vrf_name for oper writes, need to extend the peer_cfg to hold them, and also pass them when a tunnel is configured.